registry/coder/modules/agentapi/main.tf

```diff
@@ -0,0 +1,277 @@
+      mkdir -p "${local.workdir}"
+      echo "Folder created successfully."
+    fi
+    if [ -n "${local.encoded_pre_install_script}" ]; then
+      echo "Running pre-install script..."
+      echo "${local.encoded_pre_install_script}" | base64 -d > "$module_path/pre_install.sh"
+      chmod +x "$module_path/pre_install.sh"
+      "$module_path/pre_install.sh" 2>&1 | tee "$module_path/pre_install.log"
+    fi
+
+    echo "Running install script..."
+    echo "${local.encoded_install_script}" | base64 -d > "$module_path/install.sh"
+    chmod +x "$module_path/install.sh"
+    "$module_path/install.sh" 2>&1 | tee "$module_path/install.log"
+
+    # Install AgentAPI if enabled
+    if [ "${var.install_agentapi}" = "true" ]; then
+      echo "Installing AgentAPI..."
+      arch=$(uname -m)
+      if [ "$arch" = "x86_64" ]; then
+        binary_name="agentapi-linux-amd64"
+      elif [ "$arch" = "aarch64" ]; then
+        binary_name="agentapi-linux-arm64"
+      else
+        echo "Error: Unsupported architecture: $arch"
+        exit 1
+      fi
+      if [ "${var.agentapi_version}" = "latest" ]; then
+        # for the latest release the download URL pattern is different than for tagged releases
+        # https://docs.github.com/en/repositories/releasing-projects-on-github/linking-to-releases
+        download_url="https://github.com/coder/agentapi/releases/latest/download/$binary_name"
+      else
+        download_url="https://github.com/coder/agentapi/releases/download/${var.agentapi_version}/$binary_name"
+      fi
+      curl \
+        --retry 5 \
+        --retry-delay 5 \
+        --fail \
+        --retry-all-errors \
+        -L \
+        -C - \
+        -o agentapi \
+        "$download_url"
+      chmod +x agentapi
+      sudo mv agentapi /usr/local/bin/agentapi
+    fi
+    if ! command_exists agentapi; then
+      echo "Error: AgentAPI is not installed. Please enable install_agentapi or install it manually."
+      exit 1
+    fi
+
+    echo -n "${local.agentapi_start_script_b64}" | base64 -d > "$module_path/scripts/agentapi-start.sh"
+    echo -n "${local.agentapi_wait_for_start_script_b64}" | base64 -d > "$module_path/scripts/agentapi-wait-for-start.sh"
+    chmod +x "$module_path/scripts/agentapi-start.sh"
+    chmod +x "$module_path/scripts/agentapi-wait-for-start.sh"
+
+    if [ -n "${local.encoded_post_install_script}" ]; then
+      echo "Running post-install script..."
+      echo "${local.encoded_post_install_script}" | base64 -d > "$module_path/post_install.sh"
+      chmod +x "$module_path/post_install.sh"
+      "$module_path/post_install.sh" 2>&1 | tee "$module_path/post_install.log"
+    fi
+
+    export LANG=en_US.UTF-8
+    export LC_ALL=en_US.UTF-8
+
+    cd "${local.workdir}"
+    nohup "$module_path/scripts/agentapi-start.sh" true "${var.agentapi_port}" &> "$module_path/agentapi-start.log" &
+    "$module_path/scripts/agentapi-wait-for-start.sh" "${var.agentapi_port}"
+    EOT
```

Comment:
I suggest moving this to a `script.sh` file and using `templatefile()` function. It will make maintenance easy.